platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DynamicCallProblemReporting.kt

```diff
@@ -66,17 +68,38 @@ class DefaultDynamicCallProblemReporting : DynamicCallProblemReporting {
     }
 
     override fun leaveDynamicCall(entryPoint: Any) {
+        if (currentThreadState.callStack.isEmpty()) {
+            return
+        }
         val innermostCall = currentThreadState.callStack.pop()
         check(entryPoint == innermostCall.entryPoint) { "Mismatched enter-leave calls in DynamicCallProjectIsolationProblemReporting" }
     }
 
     override fun unreportedProblemInCurrentCall(problemKey: Any): Boolean {
         val currentThreadCallStack = currentThreadState.callStack
-        check(currentThreadCallStack.isNotEmpty()) { "Expected unreportedProblemInCurrentCall to be called after enterDynamicCall" }
+        if (currentThreadCallStack.isEmpty()) {
+            logMissingScope()
+            return true
+        }
+
         return currentThreadCallStack.peek().problemsReportedInCurrentCall.add(problemKey)
     }
 
+    private
+    fun logMissingScope() {
+        if (logger.isEnabled(LogLevel.DEBUG)) {
+            val stackTrace = IllegalStateException("Expected unreportedProblemInCurrentCall to be called after enterDynamicCall").stackTraceToString()
+            val shortenedStackTrace = stackTrace.lines().take(15).joinToString("\n")
+            logger.log(LogLevel.DEBUG, shortenedStackTrace)
```

Comment:
This is temporary to make sure debug logs contain enough information to track down unscoped problems.